srv/modules/runners/changed.py

```diff
@@ -16,38 +16,80 @@
 log = logging.getLogger(__name__)
 
 
-class Config(object):
+class UnknownRole(Exception):
     """
-    Tracks the configuration files, related dependencies and checksums
+    Raise when passed an unknown type or role to
+    Role().
+    """
+    pass
+
+
+class Role(object):
     """
+    Class for roles to access meta data more easily
+    """
+
+    def __init__(self, **kwargs):
+        self._role_name = kwargs.get('name')
+        self.conf_dir = kwargs.get('conf_dir', '/srv/salt/ceph/configuration/files/ceph.conf.d/')
+        self.conf_filename = kwargs.get('conf_filename', self._role_name)
+        self.conf_extension = kwargs.get('conf_extension', '.conf')
+        self._conf_files = [self.conf_dir + self.conf_filename + self.conf_extension]
+        self._depends = [self]
```

Comment:
The dependency map of 'global' should actually contain all core services (at least the mons).. Ideally we'd maintain a list of options<->service mapping and assign the correct dependencies based on that..